newstroke_hershey/convert.py

```diff
@@ -0,0 +1,154 @@
+    return res
+
+
+def format_polylines(polylines):
+    return '\n'.join([';'.join([format_vertex(v) for v in p]) for p in polylines])
+
+
+def convert_x(hershey):
+    value = ord(hershey) - ord('R')
+    value *= 9.0 / 21.0  # hershey to fontobene scaling
+    return value
+
+
+def convert_y(hershey):
+    value = ord(hershey) - ord('R')
+    value -= 9      # y offset
+    value = -value  # y invert
+    value *= 9.0 / 21.0  # hershey to fontobene scaling
+    return value
+
+
+def convert_polyline(hershey):
+    x_min = None
+    x_max = None
+    vertices = list()
+    for i in range(0, len(hershey) // 2):
+        x = convert_x(hershey[i * 2])
+        y = convert_y(hershey[i * 2 + 1])
+        vertices.append((x, y, 0))
+        x_min = min(x, x_min) if x_min else x
+        x_max = max(x, x_max) if x_max else x
+    return vertices, x_min, x_max
+
+
+def convert_polylines(hershey):
+    x_min_total = None
+    x_max_total = None
+    polylines = list()
+    for p in hershey.split(' R'):
+        if len(p):
+            polyline, x_min, x_max = convert_polyline(p)
+            polylines.append(polyline)
+            x_min_total = min(x_min, x_min_total) if x_min_total else x_min
+            x_max_total = max(x_max, x_max_total) if x_max_total else x_max
+    return polylines, x_min_total, x_max_total
+
+
+def offset_polylines(polylines, x_offset, y_offset):
+    return [[(v[0] + x_offset, v[1] + y_offset, v[2]) for v in p] for p in polylines]
+
+
+def convert_glyph(codepoint, hershey):
+    header = '[{:04X}] {}\n'.format(codepoint, chr(codepoint))
+    polylines, glyph_left, glyph_right = convert_polylines(hershey[2:])
+    if len(polylines) > 0:
+        # Actually in hershey fonts every glyph has defined its own spacing
+        # before and after the glyph. That spacing is then used by the font
+        # layout engine (no additional spacing is required). But in FontoBene
+        # we have a global letter spacing which is applied to every glyph.
+        # So we ignore the spacing of the hershey font and left-align every
+        # glyph to X=0 to not mess up spacing of FontoBene layout engines.
+        body = format_polylines(offset_polylines(polylines, -glyph_left, 0))  # left-align glyph
+    else:
+        # It seems to be a whitespace-only glyph. The whitespace width is
+        # defined by the first two letters in the hershey glyph. Because
+        # of the different concept of letter spacing, we have to subtract
+        # around 3.26 from the total width to get a suitable FontoBene
+        # spacing value. Negative spacing values are replaced by zero.
+        left = convert_x(hershey[0])
+        right = convert_x(hershey[1])
+        body = '~{}'.format(format_number(max(right - left - 3.26, 0)))
+    return header + body
+
+
+if __name__ == '__main__':
+
+    # generate C file with hershey font
+    os.system('awk -f fontconv.awk symbol.lib font.lib charlist.txt > newstroke_font.c')
```

Comment:
It would certainly be cleaner if you'd use `subprocess.check_output` and then write the output to the `newstroke_font.c` file.

On the other hand this is just a simple converter script and none of the command arguments are dynamic. So it's your choice. But at the very least you should check the return status and abort if the awk call failed.

Reply:
Yes, but then just check the return status and raise an error or exit with an error message if the status is != 0.